assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -14530,9 +14538,22 @@ else if (deleted.equalsIgnoreCase(Boolean.TRUE.toString()) && (a.getContent().ge
 			}
 			else
 			{
-				// read all Assignments
-				returnResources = assignmentService.getListAssignmentsForContext((String) state
-						.getAttribute(STATE_CONTEXT_STRING));
+				if (isSynopticTool() & isOnWorkspaceTab()) {
+					//Get all sites for this user
+					String userId = sessionManager.getCurrentSessionUserId();
+	                List<Site> sites = siteService.getSites(SelectionType.ACCESS, null, null, null,null, null);
+	                for (Site site : sites) {
+	                	Member sm = site.getMember(userId);
+	                	if (sm != null) {
+	                		returnResources.addAll(assignmentService.getListAssignmentsForContext(site.getId()));
```

Comment:
Yeah, I did this back in something like 2011 and for sure didn't think of the performance implications. I'll see what can be done about this.